src/lib/crypto/kimchi_backend/gadgets/foreign_field.ml

```diff
@@ -653,6 +653,38 @@ let check_modulus (type f) (module Circuit : Snark_intf.Run with type field = f)
 
   check_modulus_bignum_bigint (module Circuit) foreign_field_modulus
 
+(* Gadget for creating an addition or subtraction result row (Zero gate with result) *)
```

Comment:
Hmm.. this is already on the other branches.  There are also lots of other changes and it would be time consuming and painful to merge this through the other 4 branches.

Therefore, I recommend you make https://github.com/MinaProtocol/mina/pull/13821 branch the base of this branch to save time and also get all the benefit of those branches (including safety check) into this branch so we can spot anything in the test.